File: github_cpp/13/59.cpp
#include<cstdio>
#include<iostream>

using namespace std;

void bubbleSort(int arr[],int size){
	//Algorithm : Iterative
	//Time : O(n^2)
	//Space : O(1)
	for(int i=0; i<size; ++i){
		for(int j=0; j<(size-1-i); ++j){
			if(arr[j] > arr[j+1]){
				int temp = arr[j];
				arr[j] = arr[j+1];
				arr[j+1] = temp;
			}
		}
	}
}

int main(){
	int a[] = {2,5,1,0,7,8,10,9};
	
	int size = sizeof(a)/sizeof(int);
	printf("Before Sorting...\n");
	for(int i=0; i<size; ++i){
		printf("%d ",a[i]);
	}
	printf("\n");
	
	bubbleSort(a,size);
	
	printf("After Sorting...\n");
	for(int i=0; i<size; ++i){
		printf("%d ",a[i]);
	}
	printf("\n");
	
	
	return 0;
}

File: github_code_index/java/4/59.java
class insertionsort
	{public static void main(String[] args) throws java.lang.Exception
	{int array[] = {12,34,23,3423,2,3};
	System.out.println("array before sorting");
	for(int i=0;i<6;i++)
	System.out.println(array[i]);
	System.out.println("array after sorting");
	insertionSort(array);
	for(int i=0;i<6;i++)
	System.out.println(array[i]);
	
	}
	
	public static void insertionSort(int[] arr) {

      int i, j, newValue;

      for (i = 1; i < arr.length; i++) {

            newValue = arr[i];

            j = i;

            while (j > 0 && arr[j - 1] > newValue) {

                  arr[j] = arr[j - 1];

                  j--;

            }

            arr[j] = newValue;

      }

}
	
}
	